src/ipc-handlers.ts

```diff
@@ -1159,3 +1159,18 @@ export async function isFullscreen( _event: IpcMainInvokeEvent ): Promise< boole
 	const window = await getMainWindow();
 	return window.isFullScreen();
 }
+
+export async function getUserPreference( _event: IpcMainInvokeEvent, key: string ): Promise< unknown > {
```

Comment:
I'm not sure we need these two functions or the nested "preferences" key. For the "locale", there's no nested "preferences" key, so I think we should just map what we're doing for the user's "locale".